packages/signers/src/signable-message.ts

```diff
@@ -0,0 +1,17 @@
+import { TextEncoder } from 'text-encoding-impl';
+
+import { SignatureDictionary } from './types';
+
+/** Defines a message that needs signing and its current set of signatures if any. */
+export type SignableMessage = {
+    content: Uint8Array;
```

Comment:
I didn't dig at all; is it possible to spec TypedArrays as readonly somehow? Modifying the content invalidates the signatures.

Reply:
You can't do that natively on TypeScript but I found a solution from [this article](https://www.growingwiththeweb.com/2020/10/typescript-readonly-typed-arrays.html).

```ts
type TypedArrayMutableProperties = 'copyWithin' | 'fill' | 'reverse' | 'set' | 'sort';
interface ReadonlyUint8Array extends Omit<Uint8Array, TypedArrayMutableProperties> {
    readonly [n: number]: number;
}
```

However, this needs to be added to a lower-level library and used as a requirement for most functions because, rightfully, `Uint8Array` satisfies `ReadonlyUint8Array` but `ReadonlyUint8Array` does not satisfy `Uint8Array`.

In this particular case, I cannot use `ReadonlyUint8Array` because `signBytes` accepts a `Uint8Array` and not a `ReadonlyUint8Array`. I could type-cast it for now though. What do you think?

Reply:
Frustrating, but let's punt on it. It's a flourish.